TestApplications failing on Go 1.7

As can be observed on [this Travis CI build run](https://travis-ci.org/gambol99/go-marathon/builds/158021089), running all tests under Go 1.7 produces [an error](https://travis-ci.org/gambol99/go-marathon/jobs/158021092#L202-L229):

```
=== RUN   TestApplications
--- FAIL: TestApplications (0.00s)
	Error Trace:	application_test.go:265
	Error:		Received unexpected error "Marathon API error: not found"
		
	Error Trace:	application_test.go:266
	Error:		Expected value not to be nil.
		
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x48e641]

goroutine 44 [running]:
panic(0x7083c0, 0xc420014140)
	/home/travis/.gimme/versions/go1.7.linux.amd64/src/runtime/panic.go:500 +0x1a1
testing.tRunner.func1(0xc420158240)
	/home/travis/.gimme/versions/go1.7.linux.amd64/src/testing/testing.go:579 +0x25d
panic(0x7083c0, 0xc420014140)
	/home/travis/.gimme/versions/go1.7.linux.amd64/src/runtime/panic.go:458 +0x243
github.com/gambol99/go-marathon.TestApplications(0xc420158240)
	/home/travis/gopath/src/github.com/gambol99/go-marathon/application_test.go:267 +0x141
testing.tRunner(0xc420158240, 0x790d40)
	/home/travis/.gimme/versions/go1.7.linux.amd64/src/testing/testing.go:610 +0x81
created by testing.(*T).Run
	/home/travis/.gimme/versions/go1.7.linux.amd64/src/testing/testing.go:646 +0x2ec
exit status 2
FAIL	github.com/gambol99/go-marathon	0.012s
make: *** [test] Error 1
```

Under Go 1.5 and 1.6, everything is working fine.

--- a/.travis.yml
+++ b/.travis.yml
@@ -7,5 +7,7 @@
 language: go
 go:
   - 1.5
+  - 1.6
+  - 1.7
 install:
   - make test examples

--- a/application.go
+++ b/application.go
@@ -434,8 +434,13 @@ func (r *Application) String() string {
 
 // Applications retrieves an array of all the applications which are running in marathon
 func (r *marathonClient) Applications(v url.Values) (*Applications, error) {
+	query := v.Encode()
+	if query != "" {
+		query = "?" + query
+	}
+
 	applications := new(Applications)
-	err := r.apiGet(marathonAPIApps+"?"+v.Encode(), nil, applications)
+	err := r.apiGet(marathonAPIApps+query, nil, applications)
 	if err != nil {
 		return nil, err
 	}

--- a/application_test.go
+++ b/application_test.go
@@ -17,6 +17,7 @@ limitations under the License.
 package marathon
 
 import (
+	"net/url"
 	"testing"
 	"time"
 
@@ -265,6 +266,13 @@ func TestApplications(t *testing.T) {
 	assert.NoError(t, err)
 	assert.NotNil(t, applications)
 	assert.Equal(t, len(applications.Apps), 2)
+
+	v := url.Values{}
+	v.Set("cmd", "nginx")
+	applications, err = endpoint.Client.Applications(v)
+	assert.NoError(t, err)
+	assert.NotNil(t, applications)
+	assert.Equal(t, len(applications.Apps), 1)
 }
 
 func TestListApplications(t *testing.T) {

--- a/tests/rest-api/methods.yml
+++ b/tests/rest-api/methods.yml
@@ -203,6 +203,17 @@
         }
     ]
     }
+- uri: /v2/apps?cmd=nginx
+  method: GET
+  content: |
+    {
+    "apps": [
+        {
+            "args": null,
+            "cmd": "nginx"
+        }
+    ]
+    }
 - uri: /v2/apps/fake-app/restart
   method: POST
   content: |

